error_test.go

```diff
@@ -57,6 +57,22 @@ func TestAPIErrorUnmarshalJSON(t *testing.T) {
 				assertAPIErrorMessage(t, apiErr, "")
 			},
 		},
+		{
+			name:     "parse succeeds when the innerError is not exists (Azure Openai)",
+			response: `{"message": "","type": null,"param": "","code": "","status": 0,"innererror": {}}`,
+			hasError: false,
+			checkFunc: func(t *testing.T, apiErr APIError) {
+				assertAPIErrorInnerError(t, apiErr, InnerError{})
+			},
+		},
+		{
+			name:     "parse succeeds when the innerError is not InnerError struct (Azure Openai)",
+			response: `{"message": "","type": null,"param": "","code": "","status": 0,"innererror": "test"}`,
+			hasError: true,
+			checkFunc: func(t *testing.T, apiErr APIError) {
+				assertAPIErrorInnerError(t, apiErr, InnerError{})
+			},
+		},
```

Comment:
@ZeroDeng01  Please add a test to check if the innererror field is being unmarshalled as expected using the following actual response.
```
{
   "error":{
      "message":"The response was filtered due to the prompt triggering Azure OpenAI’s content management policy. Please modify your prompt and retry. To learn more about our content filtering policies please read our documentation: https://go.microsoft.com/fwlink/?linkid=2198766",
      "type":null,
      "param":"prompt",
      "code":"content_filter",
      "status":400,
      "innererror":{
         "code":"ResponsibleAIPolicyViolation",
         "content_filter_result":{
            "hate":{
               "filtered":false,
               "severity":"safe"
            },
            "self_harm":{
               "filtered":false,
               "severity":"safe"
            },
            "sexual":{
               "filtered":true,
               "severity":"medium"
            },
            "violence":{
               "filtered":false,
               "severity":"safe"
            }
         }
      }
   }
}
```